lua/starfall/instance.lua

```diff
@@ -128,9 +128,10 @@ function SF.Instance.Compile(code, mainfile, player, entity)
 		end
 	end
 
-	local serverorclientpp = instance.ppdata.serverorclient or {}
+	local serverorclientpp, owneronlypp = instance.ppdata.serverorclient or {}, instance.ppdata.owneronly or {}
 	for filename, source in pairs(code) do
 		if doNotRun[filename] then continue end -- Don't compile data files
+		if CLIENT and owneronlypp[filename] and LocalPlayer() ~= player then continue end -- Don't compile owner-only files if not owner
```

Comment:
Could this be put in the 'if' at 136? Or do we really want there to not be an entry in instance.scripts for it?

Reply:
I thought the below 'if' was weird. Why is it an empty function and instead not compiled?
Something like this where your main file is shared and it includes a serverside file:
```lua
--@shared

include("serveronly.txt")
```
Would work where it runs serverside, but clientside it will do nothing silently (because it's just an empty function).

Reply:
I feel that should throw an error clientside instead because including a serverside file from clientside doesn't really make sense.